karapace/kafka_rest_apis/__init__.py

```diff
@@ -607,28 +633,72 @@ async def get_topic_config(self, topic: str) -> dict:
         async with self.admin_lock:
             return self.admin_client.get_topic_config(topic)
 
-    async def cluster_metadata(self, topics: Optional[List[str]] = None) -> dict:
-        async with self.admin_lock:
-            if self._metadata_birth is None or time.monotonic() - self._metadata_birth > self.metadata_max_age:
-                self._cluster_metadata = None
+    def is_global_metadata_old(self) -> bool:
+        return (
+            self._global_metadata_birth is None or (time.monotonic() - self._global_metadata_birth) > self.metadata_max_age
+        )
 
-            if self._cluster_metadata:
-                # Return from metadata only if all queried topics have cached metadata
-                if topics is None:
-                    if self._cluster_metadata_complete:
-                        return self._cluster_metadata
-                elif all(topic in self._cluster_metadata["topics"] for topic in topics):
-                    return {
-                        **self._cluster_metadata,
-                        "topics": {topic: self._cluster_metadata["topics"][topic] for topic in topics},
-                    }
+    def is_metadata_of_topics_old(self, topics: list[str]) -> bool:
+        # Return from metadata only if all queried topics have cached metadata
+
+        if self._cluster_metadata_topic_birth is None:
```

Comment:
Never null, gets set in the constructor to empty dictionary.

Reply:
ops, forgot after a refactor! Thanks